### ISSUE TITLE
[BUG] Delayed version bumps after a release

### ISSUE THREAD

**Issue by gaiksaya** [opensearch-project/OpenSearch-Dashboards#7162]
**Describe the bug**

As soon as we release OpenSearch and OpenSearch-Dashboards, one of the post release activity is to bump the version in the development branches to next iteration. See https://github.com/opensearch-project/OpenSearch-Dashboards/issues/6718
However, it has been observed that many times the version is not bumped for a long amount of time. Example: 
https://github.com/opensearch-project/OpenSearch-Dashboards/blob/1.3/package.json#L14 is still pointing to 1.3.17 which was released on June 06th, 2024.

The distribution workflow in opensearch-build repository uses manifests as source of build. We have an [automation](https://github.com/opensearch-project/opensearch-build/blob/main/jenkins/manifests-update.jenkinsFile) in place, that will add those manifests when it detects that the core engine has bumped the version in tracking branches. 
However, since the versions are not bumped sooner, this delays the below actions:
1. Creating input manifest
2. Auto version increment PRs for dependent components
3. Building next iteration of OpenSearch-Dashboards distribution and help detecting failing component builds
4. Running integration tests (part of build workflow) that would catch the issues sooner

**To Reproduce**
See the version bump PR timelines in merged PRs

**Expected behavior**
Bump the version as soon as a tag is cut in this repo. This can be automated similar to OpenSearch https://github.com/opensearch-project/OpenSearch/blob/main/.github/workflows/version.yml

This would be a notable improvement to expedite the release process and avoid last minute chaos.

**Comment by joshuali925** [comment]
this doesn't seem to be a bug. what is the recommended action to improve?

i don't know about this repo, but in many other places i see version bump PR CI failed due to dependency version not upgraded

**Comment by gaiksaya** [comment]
> this doesn't seem to be a bug. what is the recommended action to improve?
> 
> i don't know about this repo, but in many other places i see version bump PR CI failed due to dependency version not upgraded

I added it as a bug since manual version bumps are not being done is timely manner. One of the recommended approach would be to parse the package.json and bump the version when a tag is cut in this repo. See OpenSearch workflow posted above. 

@AMoo-Miki @kavilla do you guys have a better approach?

**Comment by AMoo-Miki** [comment]
I believe we are relying on the human factor to make sure snapshots and possibly other artifacts catch up or else we will have a period where everything fails, blocking us from merging anything.

**Comment by AMoo-Miki** [comment]
@gaiksaya any thoughts?

**Comment by gaiksaya** [comment]
Hi @AMoo-Miki ,

Yeah that is the ask. Add an automation such that when a tag is cut version bump PR is created. Keep running CI's until they pass and once done merge it. 
I have posted a similar example in the issue description on how OpenSearch does it today.

**Comment by dbwiddis** [comment]
We are 3 days away from [2.19.2 First RC](https://opensearch.org/releases/)

[23 repos have not done their version bump](https://github.com/search?q=org%3Aopensearch-project+%5BAUTO%5D+Increment+version+to+2.19.2-SNAPSHOT&type=pullrequests).

I'm one of them and am [still waiting on upstream repos](https://github.com/opensearch-project/flow-framework/pull/1076#issuecomment-2708450819).

We have been unable to test our code against any possible upstream dependency changes since March 7.

This has been requested multiple times in the last year, including this issue and https://github.com/opensearch-project/automation-app/issues/37

What can we do to permanently fix this process?

@gaiksaya @peterzhuamazon @prudhvigodithi

**Comment by d-buckner** [comment]
Makes sense to me. Looks like we can use the [opensource-core version workflow](https://github.com/opensearch-project/OpenSearch/blob/main/.github/workflows/version.yml) as reference.

**Comment by ashwin-pc** [comment]
@sejli can you pick this up?

**Comment by gaiksaya** [comment]
Hi @sejli 
Looks like `2.19` is still pointing at 2.19.2.
https://github.com/opensearch-project/OpenSearch-Dashboards/blob/2.19/package.json#L14
It should be bumped immediately after the release. Can you please take care of it until this automation gets in?
Thanks!
cc: @ashwin-pc

**Comment by gaiksaya** [comment]
I have created a pull request to try something out to fix this. Please take a look https://github.com/opensearch-project/OpenSearch-Dashboards/pull/10112
Thanks!

**Comment by dbwiddis** [comment]
🎉

**Comment by gaiksaya** [comment]
Looks like the change worked: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/10423

**Comment by dbwiddis** [comment]
> Looks like the change worked: [#10423](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/10423)

Yes and no.

Yes the automation worked. 🎉 

However it's a sledgehammer and doesn't recognize if changes have been pushed to the PR when it retries, and it overwrites them. 😄  

Granted, it's rare, but sometimes there's an incompatible change in a version bump that requires a minor update to fix, and the overwrite behavior requires that change to be submitted in a separate PR rather than just piggy-backing onto the existing one.  (Specific example: https://github.com/opensearch-project/flow-framework/pull/1195).   Not a big deal, just a "hey this would improve this process for active maintainers who are on top of it and don't need the automation quite as much" crowd.

I really really really like the new process, and want to stress that i REALLY thank you for implementing this. This isn't a complaint, more of a feature request "let's make this better" comment.

Yeah, I know that's advanced functionality but I've come to expect that from Mend Remediate, or Dependabot.  So stick that on your "TODO" list.

**Comment by gaiksaya** [comment]
@dbwiddis The one you are referring to (basically for all plugins) these are the workflows:
OpenSearch plugins: https://github.com/opensearch-project/opensearch-build/blob/main/.github/workflows/os-increment-plugin-versions.yml
Dashboard plugins: https://github.com/opensearch-project/opensearch-build/blob/main/.github/workflows/osd-increment-plugin-versions.yml

Yes its kind of sledgehammer for those too. Maybe need to look into [pull request creation](https://github.com/opensearch-project/opensearch-build/blob/main/.github/workflows/os-increment-plugin-versions.yml#L149-L167) action that blindly overwrites the change.

For core, both OpenSearch and OpenSearch-Dashboards they have their own implementation for version bump.
